WordPress/Classes/ViewRelated/Post/WPPostViewController.m

```diff
@@ -568,8 +568,19 @@ - (void)showOnboardingTips
 
 #pragma mark - Actions
 
-- (void)showBlogSelectorPrompt
+- (void)showBlogSelectorPrompt:(WPBlogSelectorButton*)sender
 {
+    // If we are editing an existing post, we need to simply exit
+    // the screen we are currently on and return.
+    if (sender.buttonMode == WPBlogSelectorButtonSingleSite) {
+        if (self.isEditing) {
```

Comment:
This `if` block looks like a good opportunity for a separate method definition.  Something around the lines of:

```objective-c
if (sender.buttonMode == WPBlogSelectorButtonSingleSite) {
   [self cancelEditingOrDismiss];
   return;
}
```